superset/connectors/sqla/models.py

```diff
@@ -990,7 +990,9 @@ def adhoc_column_to_sqla(  # pylint: disable=too-many-locals
         time_grain = col.get("timeGrain")
         has_timegrain = col.get("columnType") == "BASE_AXIS" and time_grain
         is_dttm = False
+        pdf = None
         if col_in_metadata := self.get_column(expression):
+            pdf = col_in_metadata.python_date_format
```

Comment:
Given that you're changing how the Python date format logic is used, would you mind adding a unit test for this.